torch/lib/c10d/CMakeLists.txt

```diff
@@ -114,15 +115,15 @@ target_include_directories(c10d PUBLIC
 target_include_directories(c10d PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/..)
 
 if(USE_C10D_NCCL)
-  target_compile_definitions(c10d INTERFACE USE_C10D_NCCL)
+  target_compile_definitions(c10d PUBLIC USE_C10D_NCCL)
```

Comment:
For posterity, these have been changed from `INTERFACE` to `PUBLIC` because now we also use these compile definitions in the implementation of `c10d` itself (e.g. checking `USE_C10D_NCCL` in `frontend.cpp` to conditionally initialize nccl process group bindings).